src/connectors/CustomEventConnector.ts

```diff
@@ -0,0 +1,31 @@
+import { BaseConnector, EventConfiguration } from 'reshuffle-base-connector'
+
+export interface CustomEventConnectorEventOptions {
+  channel: string
+  payLoad?: any
+}
+
+export default class CustomEventConnector extends BaseConnector<
+  null,
+  CustomEventConnectorEventOptions
+> {
+  on(
+    options: CustomEventConnectorEventOptions,
+    handler: any,
+    eventId?: string,
+  ): EventConfiguration {
+    if (!eventId) {
+      eventId = `CustomEvent/${options.channel}/${this.id}`
+    }
+    const event = new EventConfiguration(eventId, this, options)
+    this.eventConfigurations[event.id] = event
```

Comment:
This means that you old event handler for the same string will be overrun... is that the desired behavior?

Reply:
As far as I know this is how the connectors are implemented, for the same eventId new event will be set.
I updated the fire() function to handle case when user will introduce eventId argument to the on()